pom.xml

```diff
@@ -810,6 +810,7 @@
             351
         </noSnapshot.buildvers>
         <snapshot.buildvers>
+            400
```

Comment:
This should actually only be in the snapshotScala213.buildvers at the moment. Spark 4.0.0 does not support Scala 2.12, so the CI cannot actually build the shim under the default Scala version.

Reply:
> This should actually only be in the snapshotScala213.buildvers at the moment. Spark 4.0.0 does not support Scala 2.12, so the CI cannot actually build the shim under the default Scala version.

You are right, and that is what my initial thought was but as you have noted it needs to be added to the `all.buildvers` for the 2.12 pom.xml. At this point, I have added it to the `all.buildvers` for Scala2.12 and 2.13 but as things get clearer closer to the release of Spark 4.0.0 we will have to either ignore the check for `all.buildvers` for Scala 2.12 build while keeping the check for Scala 2.13 build.

Reply:
I don't understand why 400 was added to snapshot.buildvers.  That's not what we want, right?  400 is not ready to be built.  We want 400 to be in all.buildvers but not in any definition of what is buildable.  We need 400 to be declared as a shim but not one that builds yet.  Therefore I would expect the change to be more like this:
```patch
 diff --git a/pom.xml b/pom.xml
index e898c1735a..7b45bbbd3e 100644
--- a/pom.xml
+++ b/pom.xml
@@ -849,6 +849,8 @@
             ${noSnapshot.buildvers},
             ${snapshot.buildvers},
             ${databricks.buildvers},
+           <!-- 400 is not buildable yet, only declaring it as a known shim by placing it here -->
+           400
         </all.buildvers>
         <noSnapshotScala213.buildvers>
             330,
```

Reply:
Hmm, probably something for a future PR, but we should consider how to organize this in the future for 40x shims that will only build under Scala 2.13.  These future shims (400, 401, etc.) should be able to live in `all.buildvers` but the build system can handle them as well. Maybe put it in another section that shared with `*Scala213.buildvers` sections as well?